internal/controller/services/auth/auth_controller_actions.go

```diff
@@ -143,3 +147,58 @@ func managePermissions(ctx context.Context, rr *odhtypes.ReconciliationRequest)
 
 	return nil
 }
+
+func createUserGroup(ctx context.Context, rr *odhtypes.ReconciliationRequest, userGroupName string) error {
+	namespace, err := actions.ApplicationNamespace(ctx, rr)
+	if err != nil {
+		logf.Log.Error(err, "error getting application namespace")
+		return err
+	}
+	userGroup := &userv1.Group{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: userGroupName,
+			// Otherwise it errors with  "error": "an empty namespace may not be set during creation"
+			Namespace: namespace,
+		},
+		// Otherwise is errors with "error": "Group.user.openshift.io \"odh-admins\" is invalid: users: Invalid value: \"null\": users in body must be of type array: \"null\""}
+		Users: []string{},
+	}
+
+	err = rr.Client.Create(ctx, userGroup)
```

Comment:
@zdtsw @StevenTobin is a customer supposed to be able to amend the created `Group`

Reply:
@robotmaxtron since the controller uses the `deploy` action, then I'd suggest, but not mandatory to:
- add the annotation `opendatahub.io/managed: false`
- add the `userGroup` to `ReconciliationRequest`'s resources, and remove the ad hoc `Create`

```go
err := rr.AddResources(userGroup)
if err != nil {
	return fmt.Errorf("...: %w", err)
}
```

Reply:
Thank you for the review and suggestions! As suggested, I've added the annotation and removed the Create.

Reply:
for the clean installation, Operator create default Group `odh-admin / rhods-admi`n,  this is just to matching the default group we set in the Auth CR.

afterwards, if admins want to remove `odh-admin/ rhods-admin` from Auth CR, they are allowed to delete this Group CR? 
for this case, what Operator should do upon restart or upgrade?

Reply:
In general, the path to let users delete resources create by an operator is very error prone an if that's the actual need, then the user should be the one creating and owning such resources, not the operator. 

To my understanding this PR does not change the current behavior, so, am I right assuming this question is unrelated to this specific PR ?

Reply:
you are right, it is not for this specific PR, 
only to get clarification on "An admin user should be able to add or remove users from the group yes."